File: Swift/first-unique-character-in-a-string.swift
class Solution {
    func firstUniqChar(_ s: String) -> Int {
        var dict = [Character: Bool]()

        for char in s {
            if dict[char] != nil {
                dict[char] = true
            } else {
                dict[char] = false
            }
        }

        for (i, char) in Array(s).enumerated() {
            if !dict[char]! {
                return i
            }
        }

        return -1
    }
}

File: Python/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        d = dict()

        for char in s:
            if char in d:
                d[char] = False
            else:
                d[char] = True

        for i, char in enumerate(s):
            if d[char]:
                return i

        return -1
